### PR TITLE
051523 matt eva clarifying state

### LINKED ISSUE #7: Convoluted lab that is confusing students
### Canvas Link

https://learning.flatironschool.com/courses/4331/assignments/135091?module_item_id=280802

### Concern

Defining state here as:
const [movies, setMovies] = useState({
  1: { id: 1, title: "A River Runs Through It" },
  2: { id: 2, title: "Se7en" },
  3: { id: 3, title: "Inception" },
});
is very confusing.  I think it should be:
const [movies, setMovies] = useState([
{ id: 1, title: "A River Runs Through It" },
 { id: 2, title: "Se7en" },
 { id: 3, title: "Inception" },
]);

Then in movieList they could see a clean .map
  const renderMovies = movies.map((movie) => (
    <li key={movie.id}>
      <Link to={`/movies/${movie.id}`}>{movie.title}</Link>
    </li>
  ));

I am finding students trying to following this lab in their project and getting JSON and transforming it to look like the state here and then using:
  const renderMovies = Object.keys(movies).map((movieID) => (
    <li key={movieID}>
      <Link to={`/movies/${movieID}`}>{movies[movieID].title}</Link>
    </li>
  ));
Which doesn't make any sense at all but this is the single lab that covers this topic.

### Additional Context

_No response_

### Suggested Changes

App.js:
  const [movies, setMovies] = useState([
    { id: 1, title: "A River Runs Through It" },
    { id: 2, title: "Se7en" },
    { id: 3, title: "Inception" }
  ]);

  return (
    <div>
      <NavBar />
      <Switch>
        <Route path="/movies">
          <MoviesPage movies={movies} />
        </Route>
        <Route exact path="/">
          <div>Home</div>
        </Route>
      </Switch>
    </div>
  );
}

MovieList.js:
function MoviesList({ movies }) {
  const renderMovies = movies.map((movie) => (
    <li key={movie.id}>
      <Link to={`/movies/${movie.id}`}>{movie.title}</Link>
    </li>
  ));

  return <ul>{renderMovies}</ul>;
}

And in the MovieShow.js use the id from params to .find the movie in movies

### PR DESCRIPTION
**Pull request by Matt-Eva**
Closes #7 

Hey Nancy! Thanks for raising this issue. I've implemented changes to the README that should convert this lab to use an array as state instead of an object.

I tested out the alternate code to make sure it still works the way it's intended to work, and it looks like everything is in good shape!

Thanks for catching this!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -168,18 +168,16 @@ export default MoviesList;
...
 
-The `movies` prop is an object containing each movie. To iterate over this
-object, we are using `Object.keys(movies)` to get an array of keys, then mapping
-over this array. Since the keys in the object _are also the id values for each
-movie_, the elements in `.map()` are referred to as `movieID`. We are using
-`movieID` directly in the `key` attribute, and also using it to get information
-from the `movies` object, as we see with `movies[movieID].title`.
+The `movies` prop is an array containing each movie object. We're mapping over
+this array to create an `li` element and a `Link` for each movie. We are using
+the `id` of each move as the `key` attribute and in the relative `href` for our
```
Comment (lizbur10) [inline README.md]:
```suggestion
the `id` of each movie as the `key` attribute and in the relative `href` for our
```

File: README.md [hunk 2]
```diff
@@ -168,18 +168,16 @@ export default MoviesList;
...
+`Link`s, and are using the `title` of each movie as the text for our `Link`s.
 
 In the `Link`, we've used interpolation to insert `movieID` into our path to
```
Comment (lizbur10) [inline README.md]:
```suggestion
In the `Link`, we've used interpolation to insert `movie.id` into our path to
```

File: README.md [hunk 3]
```diff
@@ -350,9 +350,13 @@ Here, we've got our `movies` as an object in props. We also have our `params`
 object which was returned from `useParams` based on the current URL. In this
```
Comment (lizbur10) [inline README.md:350]: In line 349 it should say "...as an array in props."

File: README.md [hunk 4]
```diff
@@ -434,14 +438,14 @@ to work, we needed to complete the following steps:
...
-- In `MovieShow.js`, import `useParams`; use the `:movie_id` from the params
-  object to access the correct movie from the `movies` object and display it on
-  the page
+  route, render `<MovieShow>`, passing the `movies` array as props
+- In `MovieShow.js`, import `useParams`; use the `:movieId` from the params
+  object to find the correct movie from the `movies` array and display it on the
+  page
 
 In setting up our nested routes, we made use of two hooks provided by React
 Router: `useRouteMatch` and `useParams`. The first is used to retrieve the URL
```
Comment (lizbur10) [inline README.md:451]: There are two occurrences of ':movie_id' in the Conclusion that should be updated to ':movieId'

**Review by lizbur10 — CHANGES_REQUESTED** [review]
Hi Matt - this looks great. Just a few minor updates.

File: README.md (hunk 3 above)
Comment (Matt-Eva) [reply README.md:350]: Fixed!

File: README.md (hunk 4 above)
Comment (Matt-Eva) [reply README.md:451]: Fixed!

File: README.md (hunk 2 above)
Comment (Matt-Eva) [inline README.md]:
```suggestion
In the `Link`, we've used interpolation to insert `movie.id` into our path to
```

**Pull request merged**